platform/linux-generic/drv_driver.c

```diff
@@ -186,11 +273,26 @@ void _odpdrv_driver_probe_drv_items(void)
 		enumr_c = enumr_c->next;
 	}
 	enumr_class_list_read_unlock();
+
+	/* go through the list of registered enumerator probing the new
+	 * (never probed) ones:
+	 */
+	enumr_list_read_lock();
```

Comment:
Could there be several readers executing this code simultaneously, or is it assured that only one will? If not, then this should be a write lock. If there is only one reader, consider whether this locked region is needed.